File: hashtable/409/409.js
//1
var longestPalindrome = function (s) {
  let count = {};
  let ans = 0;
  let oddCount = 0;
  for (let l of s) {
    if (!count[l]) {
      count[l] = 0;
    }
    count[l]++;
  }
  for (let freq of Object.values(count)) {
    if (freq % 2 === 0) {
      ans += freq;
    } else {
      ans += freq - 1;
      oddCount++;
    }
  }
  return oddCount > 0 ? ans + 1 : ans;
};

var longestPalindrome = function (s) {
  let set = new Set();
  for (let l of s) {
    if (set.has(l)) {
      set.delete(l);
    } else {
      set.add(l);
    }
  }
  return set.size > 0 ? s.length - set.size + 1 : s.length;
};

File: hashtable/409_longest_palindrome/409.py
class Solution:
    def longestPalindrome(self, s: str) -> int:
        ans = 0
        odd_count = 0
        for freq in collections.Counter(s).values():
            if freq % 2 == 0:
                ans += freq
            else:
                ans += freq - 1
                odd_count += 1
        return ans if odd_count == 0 else ans + 1


class Solution:
    def longestPalindrome(self, s: str) -> int:
        odd_set = set()
        for l in s:
            if l not in odd_set:
                odd_set.add(l)
            else:
                odd_set.remove(l)
        return len(s) if len(odd_set) == 0 else len(s) - len(odd_set) + 1
